### DISCUSSION TITLE
Prefer Fakes Over Mocks

### DISCUSSION THREAD

**Discussion by giscus** [Tyrrrz/Tyrrrzme#28] [Blog]
# Prefer Fakes Over Mocks

Oleksii Holub (@tyrrrz) is a software developer, open-source maintainer, tech blogger and conference speaker

https://tyrrrz.me/blog/fakes-over-mocks

**Comment by anderssv** [comment]
Excellent post! :-) I really find Fakes superior to other Test Doubles and are nagging about it left and right. If you're interested I have tried to exemplify a few of my thoughts (that are quite similar to yours) here: https://github.com/anderssv/the-example/blob/main/doc/tdd.md

**Comment by DevSavvySerge** [comment]
```c#
[Fact]
public async Task I_can_get_the_content_of_a_previously_saved_document()
{
    // Arrange
    var blobStorage = new FakeBlobStorage();
    var documentManager = new DocumentManager(blobStorage);

    await documentManager.SaveDocumentAsync("test.txt", "hello");

    // Act
    var content = await documentManager.GetDocumentAsync("test.txt");

    // Assert
    content.Should().Be("hello");
}
```

Wouldn't the (TDD) minimal implementation then be to store "hello" in a dictionary and ignore `IBlobStorage`?

**Reply by Tyrrrz** [reply]
You mean how it was shown in the previous example, in the test named `I_can_get_the_content_of_an_existing_document`?
